Spawn-App-iOS-SwiftUI/Views/SubComponents/TagsSubViews/TagsListView.swift

```diff
@@ -15,18 +15,85 @@ struct TagsListView: View {
     }
     
     var body: some View {
-        ScrollView {
-            VStack(alignment: .leading, spacing: 20) {
-                // Title and Time Information
-                ForEach(viewModel.friendTags) { friendTag in
-                    Text(friendTag.displayName)
+        VStack{
+            //search bar
+            searchView
+            
+            ScrollView {
+                //list of tags
+                VStack(alignment: .leading, spacing: 20) {
+                    // Title and Time Information
+                    ForEach(viewModel.friendTags) { friendTag in
+                        HStack {
+                            Text(friendTag.displayName)
+                                .foregroundColor(.white)
+                                .padding()
+                            Spacer()
+                            Image(systemName: "chevron.down")
+                                .foregroundColor(.white)
+                                .padding()
+                        }
+                        .background(friendTag.color)
+                        .cornerRadius(10)
+                        .padding(.horizontal)
+                    }
                 }
+                .padding(20)
+                .background(universalAccentColor)
+                .cornerRadius(universalRectangleCornerRadius)
             }
-            .padding(20)
-            .background(universalAccentColor)
-            .cornerRadius(universalRectangleCornerRadius)
+            
+            //add tag button
+            Button(action: {
+                // Action for adding a new tag
+            }) {
+                HStack {
+                    Image(systemName: "plus")
+                        .font(.title)
+                }
+                .frame(maxWidth: .infinity, minHeight: 50)
+                .overlay(
+                    RoundedRectangle(cornerRadius: 10)
+                        .stroke(style: StrokeStyle(lineWidth: 2, dash: [5]))
+                        .foregroundColor(.gray)
+                )
+            }
+            //            .padding(.horizontal) // Reduces padding on the bottom
+            //            .padding(.top, 200)
+        }
+        .padding()
+        .background(
+            RoundedRectangle(cornerRadius: 30)
+                .fill(Color(hex: "#E7E7DD"))
+                .shadow(color: .gray.opacity(0.2), radius: 10, x: 0, y: 5)
+        )
+        .padding()
+    }
+}
+
+extension TagsListView {
+    var searchView: some View {
+        VStack{
+            HStack {
+                Image(systemName: "magnifyingglass")
+                    .font(.title3)
+                    .foregroundColor(universalAccentColor)
```

Comment:
Not necessary, but a tip: 
- if you have multiple elements in a `Group`, `HStack`, or `VStack` that all have the same view modifiers applied to them, e.g. `.foregroundColor(universalAccentColor)` here, then you can simply move that out and apply it to the `Group`, `HStack`, or `VStack` itself.